From 3ab2db8e6e52767996e16d33a8672983418e448d Mon Sep 17 00:00:00 2001
From: Weiguang Kong <weiguang.kong@nxp.com>
Date: Wed, 13 Sep 2017 19:11:02 +0800
Subject: [PATCH 2530/5242] MLK-16450-2: ASoC: fsl_hifi4: free unused memory
 when closing device

commit  80894c620094c468d664e27552079887a761da20 from
https://source.codeaurora.org/external/imx/linux-imx.git

free unused memory which is allocated in fsl_hifi4_open() function
when closing the hifi4 device.

Signed-off-by: Weiguang Kong <weiguang.kong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_hifi4.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sound/soc/fsl/fsl_hifi4.c b/sound/soc/fsl/fsl_hifi4.c
index 6811dce3..826714e 100644
--- a/sound/soc/fsl/fsl_hifi4.c
+++ b/sound/soc/fsl/fsl_hifi4.c
@@ -1201,6 +1201,13 @@ static int fsl_hifi4_open(struct inode *inode, struct file *file)
 
 static int fsl_hifi4_close(struct inode *inode, struct file *file)
 {
+	struct fsl_hifi4 *hifi4_priv = dev_get_drvdata(hifi4_miscdev.parent);
+	struct device *dev = hifi4_priv->dev;
+	struct fsl_hifi4_engine *hifi4_engine;
+
+	hifi4_engine = file->private_data;
+	devm_kfree(dev, hifi4_engine);
+
 	return 0;
 }
 
-- 
1.7.9.5

